kolibri/core/decorators.py

```diff
@@ -336,12 +328,44 @@ def cache_no_user_data(view_func):
     on a per user basis.
     """
 
+    CACHE_TIMEOUT = 15
+    _response = local()
+
+    def render_and_cache(response, cache_key):
+        response.render()
+        etag = hashlib.md5(
+            kolibri_version.encode("utf-8") + str(response.content).encode("utf-8")
+        ).hexdigest()
+        cache.set(cache_key, etag, CACHE_TIMEOUT)
+        return etag
+
+    def calculate_spa_etag(*args, **kwargs):
+        # Clear the local thread 'response' property
+        setattr(_response, "response", None)
+
+        request = args[0]
+        etag = cache.get(request.path)
```

Comment:
Thought of one more thing in review:

I don't think it is huge likely, but might be good to prefix the request path with something to prevent accidental cache key collisions if something else is using the path:

So we can add this in the local scope:
`CACHE_KEY_TEMPLATE = "SPA_ETAG_CACHE_{}"`

Then to generate the cache key:
`cache_key = CACHE_KEY_TEMPLATE.format(request.path)`

Reply:
Missed using the cache key template here :)